src/utils/sformat.fs

```diff
@@ -54,65 +54,73 @@ namespace Microsoft.FSharp.Text.StructuredFormat
 
     [<NoEquality; NoComparison>]
 #if COMPILER
+    type internal BoxRange = BoxRange of obj
```

Comment:
This still makes me really uneasy. I don't know what @KevinRansom and @dsyme think of this - surely there's a way we can sort this out without having to resort to 'obj'. We're just completely subverting the type system here and saying 'trust me I know what I'm doing'.